apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -38,6 +40,16 @@ def find_ml(current, keys):
             current = current.setdefault(key, {})
     return current, keys[-1]
 
+def update(data, new_data):
+    for key in new_data:
+        if key in data:
+            if isinstance(data[key], list) and isinstance(new_data[key], list):
```

Comment:
More complete list of type testing is `if instance(plaah, (list, tuple, Sequence)) and not isinstance(plaah, (str, bytes))`

Reply:
we can probably assume that anything already in the data is a list? and do all the dict-like datatypes in collections return True on `isinstance(foo, dict)`?

Reply:
ah the dict thing was in the next comment

Reply:
For example, ChangesList doesn't inherit from a list, but it does from Sequence. Sadly, so does str.

Reply:
okay. but no tuples in the original data?

Reply:
Most likely no. Techincally tuples would be fine for `new_data`, but not for `data` (can't append to tuples)